core/src/main/java/io/github/tcdl/msb/api/Requester.java

```diff
@@ -31,6 +31,17 @@
      * @throws JsonConversionException if unable to parse message to JSON before sending to bus
      */
     void publish(Payload requestPayload);
+
+    /**
+     * Wraps a payload with protocol information, preserves original message correlationId and sends to bus.
+     * In case Requester created with expectation for responses then process them.
+     *
+     * @param requestPayload payload which will be sent to bus
+     * @param correlationId payload which will be sent to bus
+     * @throws ChannelException if an error is encountered during publishing to bus
+     * @throws JsonConversionException if unable to parse message to JSON before sending to bus
+     */
+    void publish(Payload requestPayload, String correlationId);
```

Comment:
We should have message here. correlationId in only one possible field that get propagated.